res/controllers/Numark-Mixtrack-Platinum-scripts.js

```diff
@@ -0,0 +1,1415 @@
+        midi: [0x90 + midi_chan, 0x02],
+        off: 0x01,
+        sendShifted: true,
+        shiftControl: true,
+        shiftOffset: 1,
+    });
+
+    this.pfl_button = new components.Button({
+        midi: [0x90 + midi_chan, 0x1B],
+        key: 'pfl',
+        off: 0x01,
+        type: components.Button.prototype.types.toggle,
+        connect: function() {
+            components.Button.prototype.connect.call(this);
+            this.connections[1] = engine.connectControl(this.group, this.outKey, MixtrackPlatinum.pflToggle);
+        },
+    });
+
+    this.hotcue_buttons = new components.ComponentContainer();
+    this.sampler_buttons = new components.ComponentContainer();
+    for (var i = 1; i <= 4; ++i) {
+        this.hotcue_buttons[i] = new components.HotcueButton({
+            midi: [0x94 + midi_chan, 0x18 + i - 1],
+            number: i,
+            sendShifted: true,
+            shiftControl: true,
+            shiftOffset: 8,
+        });
+
+        // sampler buttons 5-8
+        this.sampler_buttons[i] = new components.SamplerButton({
+            midi: [0x94 + midi_chan, 0x18 + i - 1],
+            sendShifted: true,
+            shiftControl: true,
+            shiftOffset: 8,
+            number: i+4,
+            loaded: 0x00,
+            playing: 0x7F,
+        });
+    }
+    this.hotcues = this.hotcue_buttons;
+
+    this.pitch = new components.Pot({
+        inKey: 'rate',
+        invert: true,
+    });
+    if (!this.active) {
+        this.pitch.firstValueReceived = true;
+    }
+
+    var pitch_or_keylock = function (channel, control, value, status, group) {
+        if (this.other.inGetValue() > 0.0 && this.isPress(channel, control, value, status)) {
+            // toggle keylock, both keys pressed
+            engine.setValue(this.group, "keylock", !engine.getValue(this.group, "keylock"));
+        }
+        else {
+            components.Button.prototype.input.call(this, channel, control, value, status, group);
+        }
+    };
+    this.pitch_bend_up = new components.Button({
+        inKey: 'rate_temp_up',
+        input: pitch_or_keylock,
+    });
+    this.pitch_bend_down = new components.Button({
+        inKey: 'rate_temp_down',
+        input: pitch_or_keylock,
+    });
+    this.pitch_bend_up.other = this.pitch_bend_down;
+    this.pitch_bend_down.other = this.pitch_bend_up;
+
+    var key_up_or_down = function (channel, control, value, status, group) {
+        this.is_pressed = this.isPress(channel, control, value, status);
+        if (this.is_pressed) {
+            if (this.other.is_pressed) {
+                // reset if both buttons are pressed
+                this.inSetValue(0.0);
+            }
+            else {
+                var old = this.inGetValue();
+                // the gui only allows 7 steps, we replicate that behavior here
+                if (Math.abs(old + this.direction) < 8.0) {
+                    this.inSetValue(Math.round(old + 1.0 * this.direction));
```

Comment:
I think this is more complicated than it needs to be. Why not just use `pitch_up/down`?

Reply:
Oh, good catch. I think that is because the `_up` and `_down` ControlPushButtons are [automatically generated](https://github.com/mixxxdj/mixxx/blob/master/src/control/controlpotmeter.cpp#L60) from the [pitch ControlPotmeter](https://github.com/mixxxdj/mixxx/blob/master/src/engine/keycontrol.cpp#L24). In this case I think it would be helpful to specifically document them though. I only found them by looking at tooltips for the buttons in Deere in Developer Mode. Could you add them to the wiki?

Reply:
lol, was hoping you would do it. I'll take care of it.

Reply:
Thanks